uni/lib/view/home/home.dart

```diff
@@ -11,11 +13,39 @@ class HomePageView extends StatefulWidget {
   State<StatefulWidget> createState() => HomePageViewState();
 }
 
-/// Tracks the state of Home page.
 class HomePageViewState extends GeneralPageViewState {
+  bool isBannerViewed = true;
+
+  @override
+  void initState() {
+    super.initState();
+    checkBannerViewed();
+  }
+
+  Future<void> checkBannerViewed() async {
+    final pref = await AppSharedPreferences.isDataCollectionBannerViewed();
+    setState(() {
+      isBannerViewed = pref;
+    });
+  }
+
+  Future<void> setBannerViewed() async {
+    await AppSharedPreferences.setDataCollectionBannerViewed(isViewed: true);
+    await checkBannerViewed();
+  }
+
```

Comment:
Aren't these functions related only to the tracking banner? Why not put it inside the banner widget?

Reply:
As `checkBannerViewed` changes the state of `HomePageView` that becomes a little more complicated, no?